rudder-web/src/main/scala/com/normation/rudder/web/rest/compliance/ComplianceAPIData.scala

```diff
@@ -0,0 +1,248 @@
+
+
+  implicit class JsonbyRuleCompliance(rule: ByRuleRuleCompliance) {
+    def toJson = (
+        ("id" -> rule.id.value)
+      ~ ("compliance" -> percents(rule.compliance))
+      ~ ("directives" -> rule.directives.map { directive =>
+          (
+              ("id" -> directive.id.value)
+            ~ ("compliance" -> percents(directive.compliance))
+            ~ ("components" -> directive.components.map { component =>
+                (
+                    ("name" -> component.name)
+                  ~ ("compliance" -> percents(component.compliance))
+                  ~ ("nodes" -> component.nodes.map { node =>
+                      (
+                          ("id" -> node.id.value)
+                        ~ ("values" -> node.values.map { value =>
+                            (
+                              ("value" -> value.componentValue)
+                                ~ ("reports" -> value.messages.map { report =>
+                                    (
+                                        ("status" -> statusDisplayName(report.reportType))
+                                      ~ ("message" -> report.message)
+                                    )
+                                })
+                            )
+                        })
+                      )
+                    })
+                )
+              })
+          )
+       })
+   )
+  }
+
+  implicit class JsonByNodeCompliance(n: ByNodeNodeCompliance) {
+    def toJson = (
+        ("id" -> n.id.value)
+      ~ ("compliance" -> percents(n.compliance))
+      ~ ("rules" -> n.nodeCompliances.map { rule =>
+          (
+              ("id" -> rule.id.value)
+            ~ ("compliance" -> percents(rule.compliance))
+            ~ ("directives" -> rule.directives.map { directive =>
+                (
+                    ("id" -> directive.id.value)
+                  ~ ("compliance" -> percents(directive.compliance))
+                  ~ ("components" -> directive.components.map { case (_, component) =>
+                      (
+                          ("name" -> component.componentName)
+                        ~ ("compliance" -> percents(component.compliance))
+                        ~ ("values" -> component.componentValues.map { case (_, value) =>
+                            (
+                              ("value" -> value.componentValue)
+                                ~ ("reports" -> value.messages.map { report =>
+                                    (
+                                        ("status" -> statusDisplayName(report.reportType))
+                                      ~ ("message" -> report.message)
+                                    )
+                                })
+                            )
+                        })
+                      )
+                    })
+                )
+             })
+         )
+        })
+    )
+  }
+
+  private[this] def statusDisplayName(r: ReportType): String = {
```

Comment:
Understanding the need it fulfills, I agree with them!

And i do not find a better way to name them and provide the same usefullness